components/dialog/RotateEncryptionKeyDialog.vue

```diff
@@ -26,25 +28,31 @@ export default {
   async fetch() {
     const c = this.cluster;
 
-    if ( c.isRke1 && this.$store.getters['isRancher'] ) {
-      let etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
+    let etcdBackups = [];
 
+    if ( c.isRke1 && this.$store.getters['isRancher'] ) {
+      etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
       etcdBackups = etcdBackups.filter(backup => backup.clusterId === c.metadata.name);
+    }
 
-      const backups = sortBy(etcdBackups, ['created']).reverse();
-      const dateFormat = escapeHtml(this.$store.getters['prefs/get'](DATE_FORMAT));
-      const timeFormat = escapeHtml( this.$store.getters['prefs/get'](TIME_FORMAT));
+    if (c.isRke2 && this.$store.getters['isRancher']) {
+      etcdBackups = await this.$store.dispatch('management/findAll', { type: SNAPSHOT });
+      etcdBackups = etcdBackups.filter(backup => backup.clusterId === c.id);
+    }
 
-      if (backups.length > 0) {
-        const name = backups[0].id.split(':');
-        const d = day(backups[0].created).format(dateFormat);
-        const t = day(backups[0].created).format(timeFormat);
+    const backups = sortBy(etcdBackups, ['created']).reverse();
+    const dateFormat = escapeHtml(this.$store.getters['prefs/get'](DATE_FORMAT));
+    const timeFormat = escapeHtml( this.$store.getters['prefs/get'](TIME_FORMAT));
 
-        this.latestBackup = {
-          name: name[0],
-          date: `${ d } ${ t }`,
-        };
-      }
+    if (backups.length > 0) {
+      const name = backups[0].id.split(':');
+      const d = day(backups[0].created).format(dateFormat);
```

Comment:
This whole computation can be wrapped in a single function, maybe as helper if the operation is recurrent:

```ts
      const formatPref = pref => {
        // we may want to avoid 1 line for readability and allow to dedbug
        const format = escapeHtml(this.$store.getters['prefs/get'](pref));
        const pref = day(backups[0].created).format(timeFormat);

        return pref;
      }

      this.latestBackup = {
        name: name[0],
        date: `${ formatPref(DATE_FORMAT) } ${ formatPref(TIME_FORMAT) }`, // I think it's still readable for debugging purposes
      };
```

Reply:
I have put the formatting logic in a separate function by rewriting it this way:

```
const getFormattedCreatedDate = (createdDate) => {
      const dateFormat = escapeHtml(this.$store.getters['prefs/get'](DATE_FORMAT));
      const timeFormat = escapeHtml( this.$store.getters['prefs/get'](TIME_FORMAT));
      const d = day(createdDate).format(dateFormat);
      const t = day(createdDate).format(timeFormat);
      return `${ d } ${ t }`;
    }

    if (etcdBackups.length > 0) {
      const backups = sortBy(etcdBackups, ['created']).reverse();
      const name = backups[0].id.split(':');
      const createdDate = backups[0].created;

      this.latestBackup = {
        name: name[0],
        date: getFormattedCreatedDate(createdDate)
      };
    }
```